application/common/models/Password.php

```diff
@@ -0,0 +1,152 @@
+                'skipOnError' => false,
+                'message' => \Yii::t(
+                    'app',
+                    'Your password exceeds the maximum length of {maxLength} (code 110)',
+                    ['maxLength' => $this->config['maxLength']['value']]
+                ),
+                'when' => function() { return $this->config['maxLength']['enabled']; }
+            ],
+            [
+                'password', 'match', 'pattern' => $this->config['minNum']['phpRegex'],
+                'skipOnError' => false,
+                'message' => \Yii::t(
+                    'app',
+                    'Your password must contain at least {minNum} numbers (code 120)',
+                    ['minNum' => $this->config['minNum']['value']]
+                ),
+                'when' => function() { return $this->config['minNum']['enabled']; }
+            ],
+            [
+                'password', 'match', 'pattern' => $this->config['minUpper']['phpRegex'],
+                'skipOnError' => false,
+                'message' => \Yii::t(
+                    'app',
+                    'Your password must contain at least {minUpper} upper case letters (code 130)',
+                    ['minUpper' => $this->config['minUpper']['value']]
+                ),
+                'when' => function() { return $this->config['minUpper']['enabled']; }
+            ],
+            [
+                'password', 'match', 'pattern' => $this->config['minSpecial']['phpRegex'],
+                'skipOnError' => false,
+                'message' => \Yii::t(
+                    'app',
+                    'Your password must contain at least {minSpecial} special characters (code 140)',
+                    ['minSpecial' => $this->config['minSpecial']['value']]
+                ),
+                'when' => function() { return $this->config['minSpecial']['enabled']; }
+            ],
+            [
+                'password', ZxcvbnPasswordValidator::className(), 'minScore' => $this->config['zxcvbn']['minScore'],
+                'skipOnError' => true,
+                'message' => \Yii::t(
+                    'app',
+                    'Your password does not meet the minimum strength of {minScore} (code 150)',
+                    ['minScore' => $this->config['zxcvbn']['minScore']]
+                ),
+                'when' => function() { return $this->config['zxcvbn']['enabled']; }
+            ],
+        ];
+    }
+
+    /**
+     * Shortcut method to initialize a Password object
+     * @param string $newPassword
+     * @return Password
+     */
+    public static function create($newPassword)
+    {
+        $password = new Password();
+        $password->password = $newPassword;
+
+        return $password;
+    }
+
+    /**
+     * @param int $userId
+     * @throws BadRequestHttpException
+     * @throws ServerErrorHttpException
+     */
+    public function save($userId)
```

Comment:
It may also be worth using `$employeeId` rather than `$userId` within the Password class, since that is what the PasswordStoreInterface uses, as well as to avoid confusion of which property to use when calling this from the User class.